core-bundle/src/Config/Loader/PhpFileLoader.php

```diff
@@ -44,71 +48,82 @@ public function supports($resource, $type = null): bool
-                    // remove
-                    break;
-
-                case false !== $buffer:
-                    $buffer .= $token;
-
-                    if (';' === (string) $token) {
-                        $code .= $this->handleDeclare($buffer);
-                        $buffer = false;
-                    }
-                    break;
-
-                case $token instanceof \PHP_Token_NAMESPACE:
-                    if ('{' === $token->getName()) {
-                        $namespace = false;
-                        $code .= $token;
-                    } else {
-                        $namespace = $token->getName();
-                        $stream->seek($token->getEndTokenId());
+        // Parse input into an AST.
+        $contents = trim(file_get_contents($file));
+        $parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+        $ast = $parser->parse($contents);
+
+        // Traverse and modify the AST.
+        $namespaceResolver = new NameResolver();
+
+        $nodeStripper = new class() extends NodeVisitorAbstract {
+            public function leaveNode(Node $node)
+            {
+                // Drop namespace and use declarations.
+                if ($node instanceof Node\Stmt\Namespace_) {
+                    return $node->stmts;
+                }
+
+                if ($node instanceof Node\Stmt\Use_) {
+                    return NodeTraverser::REMOVE_NODE;
+                }
+
+                // Drop 'strict_types' definition.
+                if ($node instanceof Node\Stmt\Declare_) {
+                    foreach ($node->declares as $key => $declare) {
+                        if ('strict_types' === $declare->key->name) {
+                            unset($node->declares[$key]);
+                        }
                     }
-                    break;
 
-                case $token instanceof \PHP_Token_DECLARE:
-                    $buffer = (string) $token;
-                    break;
+                    if (empty($node->declares)) {
+                        return NodeTraverser::REMOVE_NODE;
+                    }
+                }
+
+                // Drop any inline HTML.
+                if ($node instanceof Node\Stmt\InlineHTML) {
+                    return NodeTraverser::REMOVE_NODE;
+                }
+
+                // Drop legacy access check ("if(*) die('You cannot access this file directly!');").
+                if ($node instanceof Node\Stmt\If_) {
+                    foreach ($node->stmts as $statement) {
+                        if (
+                            $statement instanceof Node\Stmt\Expression &&
+                            $statement->expr instanceof Node\Expr\Exit_ &&
+                            $statement->expr->expr instanceof Node\Scalar\String_ &&
+                            $this->matchLegacyCheckText($statement->expr->expr->value)
+                        ) {
+                            return NodeTraverser::REMOVE_NODE;
```

Comment:
Wouldn’t this remove all `if` statements that have `die('You cannot access this file directly!')` as one of their sub-statements?

I think we should check for `TL_ROOT` too and make sure that the `die()` statement is the first one.

Reply:
Checking against all parts in 917fb7548a317443010c87e74fdcddc5ec03338b